### PR TITLE
MAINT: fix interaction with external logging

### LINKED ISSUE #2569: Importing nipype changes external logging configuration
### Summary
If you have a python application that uses logging, importing nipype modifies your logging configuration. This results in multiple log messages being printed to the screen when you go to use the logger that you set up.

### Actual behavior
Importing from nipype.interfaces.ants import N4BiasFieldCorrection reaults in log messages being printed to the console twice.

### Expected behavior
When you import nipype, it should have no effect on any logging system in your application

### How to replicate the behavior
```
import logging
from nipype.interfaces.ants import N4BiasFieldCorrection
logger = logging.getLogger('root')
logger.addHandler(logging.StreamHandler())
logger.warning("This should really only be appearing once...")
```

### Script/Workflow details
N/a
Please put URL to code or code here (if not too long).

### Platform details:
MacOS 
Please paste the output of: `python -c "import nipype; print(nipype.get_info()); print(nipype.__version__)"`

{'pkg_path': '/Users/jonpdeaton/anaconda3/lib/python3.6/site-packages/nipype', 'commit_source': 'installation', 'commit_hash': '7d3966f', 'nipype_version': '1.0.3', 'sys_version': '3.6.4 |Anaconda custom (64-bit)| (default, Mar 12 2018, 20:05:31) \n[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]', 'sys_executable': '/Users/jonpdeaton/anaconda3/bin/python', 'sys_platform': 'darwin', 'numpy_version': '1.14.2', 'scipy_version': '0.19.1', 'networkx_version': '2.0', 'nibabel_version': '2.2.1', 'traits_version': '4.6.0'}
1.0.3

### Execution environment
Anaconda, Python 3.6

Choose one
- Container [Tag: ???]
- My python environment inside container [Base Tag: ???]
- My python environment outside container

### PR DESCRIPTION
**Pull request by mgxd**
Fixes #2569  .

Changes proposed in this pull request
- remove call to `logging.basicConfig`
- scope sub-loggers (interface, workflow, etc) within `nipype` ancestor

### CODE REVIEW

File: nipype/interfaces/io.py [hunk 1]
```diff
@@ -513,7 +513,7 @@ def _fetch_bucket(self, bucket_name):
 
         # Init variables
         creds_path = self.inputs.creds_path
-        iflogger = logging.getLogger('interface')
+        iflogger = logging.getLogger('nipype.interface')
```
Comment (effigies) [inline nipype/interfaces/io.py]: I would actually remove this line, as well as the others (L590, L658), as they're overriding a file-level variable, but not changing it.

Comment (mgxd) [reply nipype/interfaces/io.py]: nice catch, fixed.

**Comment by effigies** [comment]
Is there some way to test this automatically? Or at least post a demonstration that @jondeaton's replicating code now works as expected?

**Comment by mgxd** [comment]
@effigies any recommended way to check logging output? I mocked up an example with `unittest.Testcase.assertLogs` but was passing for both pre/post fix.

@jondeaton I tested this with your minimal example
```python
import logging
from nipype.interfaces.ants import N4BiasFieldCorrection
logger = logging.getLogger('root')
logger.addHandler(logging.StreamHandler())
logger.warning("This should really only be appearing once...")
```
but perhaps you would like to check out this branch and verify this fix.

**Comment by effigies** [comment]
> @effigies any recommended way to check logging output? I mocked up an example with unittest.Testcase.assertLogs but was passing for both pre/post fix.

No, sorry, I don't have any experience here. It would just be nice to have some confirmation.

**Pull request merged**